### PR TITLE
Change block size dynamically.  #2017

### PR DESCRIPTION
**Pull request by DailyDreaming**
@ejacox I have no way of testing this, but does this fix the issue?

### CODE REVIEW

**Comment by ejacox** [comment]
@DailyDreaming, this didn't fix the problem.

**Comment by joelarmstrong** [comment]
This is actually a problem with the test, which is incorrectly calculating the filesystem's block size:

https://github.com/BD2KGenomics/toil/blob/f462225e5ddf2848ede783f953f18fe18ccc8ff3/src/toil/test/src/miscTests.py#L57

The actual getDirSizeRecursively code was probably fine the way it was. But `st_blksize` is *not* guaranteed to be the filesystem-level blocksize (i.e. the minimum allocation size). It happens to be that way on ext2/3/4 on Linux, as far as I can tell, but according to the documentation it is just the "preferred" I/O size. On macOS it's 4MiB.

If we really, really want the filesystem-level blocksize, we could think about using the `f_frsize` entry of [statvfs](http://man7.org/linux/man-pages/man3/statvfs.3.html). But why not just use du in the test to get the actual size on disk, and call it a day? I'm not sure we need to do the allocation-size guessing that the test currently does.

**Comment by DailyDreaming** [comment]
Alright, changed it to use du!

File: src/toil/common.py [hunk 1]
```diff
@@ -543,7 +543,7 @@ def _addOptions(addGroupFn, config):
     #Debug options
     #
     addOptionFn = addGroupFn("toil debug options", "Debug options")
-    addOptionFn("--debug-worker", default=False, action="store_true", dest="debugWorker",
+    addOptionFn("--debugWorker", default=False, action="store_true",
```
Comment (joelarmstrong) [inline src/toil/common.py]: Why this change? is this because of a bad merge or something? This line got changed 3 times in 4 commits! And the change has already been applied on master. IMO this PR should be squashed or otherwise rebased so this line won't be changed.

File: src/toil/common.py [hunk 2]
```diff
@@ -1257,33 +1257,26 @@ def cacheDirName(workflowID):
...
-    :rtype: int
+    This method will return a 'subprocess.CalledProcessError' if it is unable to
+    access a folder or file because of permissions.  Therefore this method should
+    only be called on the jobStore, and will alert the user if some portion is
+    inaccessible.  Everything in the jobStore should have appropriate permissions
+    as there is no way to read the filesize without permissions.
+
+    du is often faster than using os.lstat(), sometimes significantly so.
+
+    The call: 'du -c /some/path' should give the number of 512-byte blocks
```
Comment (joelarmstrong) [inline src/toil/common.py]: this still reads `du -c` instead of `du -s`

**Comment by diekhans** [comment]
seems like it mostly comments that need fixed ;-)

File: src/toil/common.py (hunk 1 above)
Comment (DailyDreaming) [reply src/toil/common.py]: Yes, it was a bad merge.  I'll just squash this and make a new one.
